### PR TITLE
Logic from index.ts to seperate utility functions

### PR DESCRIPTION
**Pull request by TomWindels**
Separated the processing logic from index.ts, allowing for easier reuse, small project restructuring and a few minor optimizations

### CODE REVIEW

File: EventSource/src/util/Processing.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ * 
+ * @param filepath
+ * @returns {Promise<Store | null>}
+ */
+export async function storeFromFile(filepath: string): Promise<Store | null> {
+    if (existsSync(filepath)) {
+        return await turtleStringToStore(readFileSync(filepath, "utf-8"));
+    }
+    return null
+}
```
Comment (woutslabbinck) [inline EventSource/src/util/Processing.ts:18]:
```suggestion
export async function storeFromFile(filepath: string): Promise<Store> {
    if (existsSync(filepath)) {
        return await turtleStringToStore(readFileSync(filepath, "utf-8"));
    }
    return new Store()
}
```
Does it make sense to have an empty store returned here?

File: EventSource/src/util/Processing.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ export function extractTimeFrom(resource: Resource, treePath: string): number {
+    // as the shape of a resource can vary, a flexible approach using a store is used
+    return extractTimestampFromLiteral(
+        new Store(resource).getObjects(
+            // at this time, all quads in the resource should use the same subject, so
+            // subject of the first quad suffices
+            resource[0].subject, treePath, null
+        )[0] as Literal
+    );
+}
```
Comment (woutslabbinck) [inline EventSource/src/util/Processing.ts:94]: Isn't this the exact same function as https://github.com/woutslabbinck/SolidEventSourcing/blob/main/EventSource/src/EventSourceUtil.ts#L55-L58

Could you make sure that in this PR only one of those functions remain?

**Review by woutslabbinck — CHANGES_REQUESTED** [review]

File: EventSource/src/util/Processing.ts (hunk 1 above)
Comment (TomWindels) [reply EventSource/src/util/Processing.ts:18]: I personally feel like passing in an invalid filepath resulting in `null` makes more sense than an empty store (which would be the result of a valid but empty file), but I can change it to return a new store instead if you prefer.

File: EventSource/src/util/Processing.ts (hunk 2 above)
Comment (TomWindels) [reply EventSource/src/util/Processing.ts:94]: This does seem to be the case. I'll remove it and change the references to use that one instead.

**Review by woutslabbinck — APPROVED** [review]

**Pull request merged**